spec/models/course_spec.rb

```diff
@@ -229,5 +229,38 @@
         it { is_expected.to be_truthy }
       end
     end
+
+    describe '.active_in_past_7_days' do
+      let!(:courses) { create_list(:course, 3) }
+
+      it 'returns the active courses' do
+        old_count = instance.courses.active_in_past_7_days.count
+
+        create(:course_user, course: courses.sample, last_active_at: Time.zone.now)
+
+        new_count = instance.courses.active_in_past_7_days.count
+        expect(new_count).to eq(old_count + 1)
+      end
+    end
+
+    describe 'calculated attributes' do
+      let(:course) { create(:course) }
+      let!(:course_users) { create_list(:course_user, 2, course: course) }
+
+      describe '.active_user_count' do
+        before { course_users.sample.update_column(:last_active_at, Time.zone.now) }
```

Comment:
The rest of the volations won't fix, `update_column` gives much better performane than `save`